### PR TITLE
Archer v2 - Add levelId arg

### PR DESCRIPTION
**Pull request by adi88d**
update docker image

## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/36114

## Description
Added the optional argument **levelId** to the commands:
- archer-create-record
- archer-search-records
- archer-update-record

in order to let the user the option to use different level Id instead of the first level of the application.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 5.5.0
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/ArcherRSA/Integrations/ArcherV2/ArcherV2.py [hunk 1]
```diff
@@ -851,12 +851,18 @@ def get_record_command(client: Client, args: Dict[str, str]):
...
+    levels = client.get_levels_by_app_id(app_id)
+
+    level_data = None
+    if level_id:
+        level_data = next((level for level in levels if level.get('level') == int(level_id)), None)
+    elif levels:
+        level_data = levels[0]
+
+    if not level_data:
+        raise DemistoException(f'Got no level by app id. You might be using the wrong application id or level id.')
```
Comment (yaakovi) [inline Packs/ArcherRSA/Integrations/ArcherV2/ArcherV2.py]: let's move this block to a function

File: Packs/ArcherRSA/Integrations/ArcherV2/ArcherV2.yml [hunk 2]
```diff
@@ -214,6 +214,12 @@ script:
       name: fieldsToValues
       required: true
       secret: false
+    - default: false
+      description: The Level ID to use for create the record.
```
Comment (yaakovi) [inline Packs/ArcherRSA/Integrations/ArcherV2/ArcherV2.yml:218]: add a note that by default the command will use the first level.

File: Packs/ArcherRSA/Integrations/ArcherV2/ArcherV2.py [hunk 3]
```diff
@@ -325,36 +325,45 @@ def do_soap_request(self, command, **kwargs):
...
+                            fields[field_id] = {'Type': field_item.get('Type'),
+                                                'Name': field_item.get('Name'),
+                                                'FieldId': field_id,
+                                                'IsRequired': field_item.get('IsRequired', False),
+                                                'RelatedValuesListId': field_item.get('RelatedValuesListId')}
+
+                    levels.append({'level': level_id, 'mapping': fields})
+            if levels:
+                cache[int(app_id)] = levels
+                demisto.setIntegrationContext(cache)
```
Comment (yaakovi) [inline Packs/ArcherRSA/Integrations/ArcherV2/ArcherV2.py:355]:
```suggestion
                set_integration_context(cache)
```

File: Packs/ArcherRSA/Integrations/ArcherV2/ArcherV2.py [hunk 4]
```diff
@@ -325,36 +325,45 @@ def do_soap_request(self, command, **kwargs):
         self.destroy_token(token)
         return extract_from_xml(res, req_data['outputPath']), res
 
-    def get_level_by_app_id(self, app_id):
+    def get_level_by_app_id(self, app_id, specify_level_id=None):
+        levels = []
         cache = demisto.getIntegrationContext()
```
Comment (yaakovi) [inline Packs/ArcherRSA/Integrations/ArcherV2/ArcherV2.py:330]:
```suggestion
        cache = get_integration_context()
```

**Review by yaakovi — APPROVED** [review]

**Comment by ShirleyDenkberg** [comment]
@yaakovi Doc review completed.

**Pull request merged**